### DISCUSSION TITLE
How to handle function with floating point arguments?

### DISCUSSION THREAD

**Discussion by TerryGuo** [GaloisInc/saw-script#2137] [Q&A]
Hi there,

I am new to SAW and formal verification, but the [tutorial](https://saw.galois.com/tutorial.html) is great, especially the part related to prove ffs_ref/ffs_imp/ffs_musl are equivalent. However for my case, the input argument is float, for example I want to prove below two C functions are equivalent:

```
float triple_f_a(float a) {
   return a + a + a;
}

float triple_f_b(float a) {
  return a * 3;
}
```
with below SAW script
```
triple_f_a <- llvm_extract a "triple_f_a"
triple_f_b <- llvm_extract a "triple_f_b"
```

When run this SAW script file, I got error:

> Extracting implementation term: triple_f_a
> Stack trace:
> "llvm_extract"(...)
> Crucible extraction currently only supports Crucible base types FloatRepr SingleFloatRepr

Sorry for my little knowldge on SAW. Any help or guide are welcomed.

**Comment by RyanGlScott** [comment]
Unfortunately, SAW is not currently capable of proving properties involving float-point types. See https://github.com/GaloisInc/saw-script/issues/1237.

**Comment by sauclovian-g** [comment]
That error message is all wet, though. We can at least improve that...

**Comment by TerryGuo** [comment]
Thanks for your helping. I will seek other tools to verify floating point type.
